src/ToricVarieties/NormalToricVarieties/methods.jl

```diff
@@ -1,21 +1,39 @@
 ############################
-# Dimensions
+# Betti numbers
 ############################
 
 @doc Markdown.doc"""
-    ith_betti_number(v::AbstractNormalToricVariety, i::Int)
+    betti_number(v::AbstractNormalToricVariety, i::Int)
 
 Compute the i-th Betti number of the normal toric variety `v`.
 """
-function ith_betti_number(v::AbstractNormalToricVariety, i::Int)
-    if isodd(i)
+function betti_number(v::AbstractNormalToricVariety, i::Int)
+    # check input
+    if i > 2*dim(v) || i < 0
         return 0
     end
-    k = div(i, 2)
-    f_vector = Vector{Int}(pm_object(v).F_VECTOR)
-    pushfirst!(f_vector, 1)
-    betti_number = sum((-1)^(i-k) * binomial(i,k) * f_vector[dim(v) - i + 1] for i=k:dim(v))
-    return betti_number
     
+    # extract vector of currently-known Betti numbers (or create it if necessary)
+    if !has_attribute(v, :betti_number)
+        betti_numbers = fill(fmpz(-1),2*dim(v)+1)
+    else
+        betti_numbers = get_attribute(v, :betti_number)
```

Comment:
To improve type stability, you could have to rewrite this as
```suggestion
        betti_numbers = get_attribute(v, :betti_number)::Vector{fmpz}
```
Alas, I'd actually instead rewrite this function to use `get_attribute!`, and I'd call the attribute `:betti_numbers` (so plural).

Reply:
Sure, let me make the corresponding changes.